skip tests if python package versions fail

milljm opened the pull request:
Allow the TestHarness to skip tests if required python package versions do not meet specified criteria.

Closes #24736

Commit:
skip tests if python package versions fail

Allow the TestHarness to skip tests if required python package
versions do not meet specified criteria.

Closes #24736

Commit:
Fix the caveat printing a bit

Make the caveats easier to read.

Commit:
Try to catch more possible parsing errors

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1596590/) on 67afc77 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24737/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1596608/) on 67afc77 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>20e735</th>
      <th colspan=4>#24737 67afc7</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.41%</td>
      <td>85.40%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>88937</td>
      <td>88932</td>
      <td>-5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15196</td>
      <td>15201</td>
      <td>+5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24737/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24737/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24737/coverage/xfem)

</details>



This comment will be updated on new commits.

Commit:
Remove packaging requirement

Remove packaging requirement from mooseutils, and create our own.
Some of our build machines do not have the Python package: packaging
available.

Normally we would "SKIP" a test which requires a Python package not
being available. But in the case where we were using this package for
testing whether or not other python packages are available, we must
instead do it ourselves.

loganharbour commented:
OpenMPI test failure is unrelated and intermittent.
